src/pal/src/eventprovider/lttng/eventprovdotnetruntimerundown.cpp

```diff
@@ -0,0 +1,1576 @@
+        MethodDCEnd_V1,
+        MethodID,
+        ModuleID,
+        MethodStartAddress,
+        MethodSize,
+        MethodToken,
+        MethodFlags,
+        ClrInstanceID
+        );
+
+        Error = ERROR_SUCCESS;
+
+return Error;
+}
+extern "C" BOOL  EventEnabledMethodDCEnd_V2(){ return TRUE;}
+extern "C" ULONG  FireEtwMethodDCEnd_V2(
+                    const unsigned __int64  MethodID,
+                    const unsigned __int64  ModuleID,
+                    const unsigned __int64  MethodStartAddress,
+                    const unsigned int  MethodSize,
+                    const unsigned int  MethodToken,
+                    const unsigned int  MethodFlags,
+                    const unsigned short  ClrInstanceID,
+                    const unsigned __int64  ReJITID
+)
+{
+ULONG Error = ERROR_WRITE_FAULT;
+                                
+     tracepoint(
+        DotNETRuntimeRundown,
+        MethodDCEnd_V2,
+        MethodID,
+        ModuleID,
+        MethodStartAddress,
+        MethodSize,
+        MethodToken,
+        MethodFlags,
+        ClrInstanceID,
+        ReJITID
+        );
+
+        Error = ERROR_SUCCESS;
+
+return Error;
+}
+extern "C" BOOL  EventEnabledMethodDCStartVerbose(){ return TRUE;}
+extern "C" ULONG  FireEtwMethodDCStartVerbose(
+                    const unsigned __int64  MethodID,
+                    const unsigned __int64  ModuleID,
+                    const unsigned __int64  MethodStartAddress,
+                    const unsigned int  MethodSize,
+                    const unsigned int  MethodToken,
+                    const unsigned int  MethodFlags,
+                    PCWSTR  MethodNamespace,
+                    PCWSTR  MethodName,
+                    PCWSTR  MethodSignature
+)
+{
+ULONG Error = ERROR_WRITE_FAULT;
+    INT MethodNamespace_path_size = 0;
+    CHAR MethodNamespace_full_name[ MAX_PATH + 1 ];
+    MethodNamespace_path_size = WideCharToMultiByte( CP_ACP, 0, MethodNamespace, -1, MethodNamespace_full_name,
+                                             MAX_PATH, NULL, NULL );
+    if( MethodNamespace_path_size == 0 ){ return ERROR_INVALID_PARAMETER;}
+    _ASSERTE(MethodNamespace_path_size == PAL_wcslen(MethodNamespace) + 1); 
+    INT MethodName_path_size = 0;
+    CHAR MethodName_full_name[ MAX_PATH + 1 ];
+    MethodName_path_size = WideCharToMultiByte( CP_ACP, 0, MethodName, -1, MethodName_full_name,
+                                             MAX_PATH, NULL, NULL );
+    if( MethodName_path_size == 0 ){ return ERROR_INVALID_PARAMETER;}
+    _ASSERTE(MethodName_path_size == PAL_wcslen(MethodName) + 1); 
+    INT MethodSignature_path_size = 0;
+    CHAR MethodSignature_full_name[ MAX_PATH + 1 ];
+    MethodSignature_path_size = WideCharToMultiByte( CP_ACP, 0, MethodSignature, -1, MethodSignature_full_name,
+                                             MAX_PATH, NULL, NULL );
+    if( MethodSignature_path_size == 0 ){ return ERROR_INVALID_PARAMETER;}
```

Comment:
Can we re-order the assert and if check so that the assert fires if the string could not be copied to the fixed buffer?  That will ensure that it's easier to tell when we need to change string handling or increase the buffer size.